sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/MapData.scala

```diff
@@ -19,6 +19,10 @@ package org.apache.spark.sql.catalyst.util
 
 import org.apache.spark.sql.types.DataType
 
+/**
+ * `MapData` should not implement `equals` and `hashCode` because the type cannot be used as join
```

Comment:
it's not your fault, but I think we need to add some comment for `MapData` itself, and then follows this comment as a note. We can simply say: `An internal data representation for map type in Spark SQL`.